src/Faker/Provider/HtmlLorem.php

```diff
@@ -129,25 +181,56 @@ private function addRandomAttribute(\DOMElement $node)
                 $node->setAttribute("class", $this->generator->word);
                 break;
             case 2:
-                $node->setAttribute("id", (string)$this->idGenerator->randomNumber(5));
+                $this->addUniqueIdAttribute($node);
+        }
+    }
+
+    /**
+     * @param \DOMElement $node
+     * @return void
+     */
+    private function addUniqueIdAttribute(\DOMElement $node)
+    {
+        for ($x = 1; $x <= 10000; $x++) {
```

Comment:
Better use a `while` than a `for` that you'll `break`. Oh, and you're free to use recursion by the way.